src/main/UI/PasswordGeneratorBox.java

```diff
@@ -0,0 +1,134 @@
+package UI;
+
+import controller.PasswordGeneratorUIControl;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.Objects;
+
+public class PasswordGeneratorBox implements ActionListener {
+    DialogBox dialogBox;
+    ComboBox comboBox;
+    TextArea textArea;
+    Button button;
+    Label label;
+    RadioBox lcbox, ucbox, numbox, spcbox;
+    Icon icon;
+
+
+    public void createPasswordGeneratorBox() {
```

Comment:
This method is huge, maybe refactor to helper methods and classes?

Reply:
The password box contains several elements which have to be instantiated & used. I have already used helper functions to define individual Ui components. I need to at least instantiate and call them. Please provide more insight.

Reply:
If you move the setting of the background and foreground colours into those UI element constructors that should help somewhat (and fix the issue of needing to access the wrapped radioButton attribute too). Also, pulling out all of the code for creating the RadioBoxes, for example, would somewhat help reduce the method length. (i.e. making a "create radio buttons" helper method and calling it to hide the code for actually setting up the RadioBoxes.)